### ISSUE TITLE
the proxy function is very useful

### ISSUE THREAD

**Issue by xiaozhu1337** [mtxr/vscode-sqltools#483]
**Is your feature request related to a problem? Please describe.**
some database need proxy to connect to it.

**Describe the solution you'd like**
add socks5 proxy support

**Describe alternatives you've considered**

**Additional context**

**Comment by jelez** [comment]
yes would be very useful when connecting to cloud databases via SSH Dynamic tunnel

**Comment by tiny1990** [comment]
+1

**Comment by mtxr** [comment]
@jelez for SSH we have #395 we it's being tracked.

@xiaozhu1337 could you give an real-world example so I can search more about your needs?

For some drivers socks connection are supported already, but I'm not sure if that's the proxy thing you are looking for

**Comment by gullpavon** [comment]
@mtxr An example would be when you're trying to connect to Snowflake (which requires connecting to somevaluehere.snowflakecomputing.com) from your company and in order to do so you need to use a proxy  i.e. proxy.c.domain.com:8080 because you are behind a firewall.

**Comment by kzajaczkowski** [comment]
+1 to this one.

Many organisations use mandatory proxy which means that you cannot connect without proxy support at all.

**Comment by betizad** [comment]
even though I have set proxy settings in my VSCODE, sqltools is not taking the proxy settings. It's very important as almost all the organization are behind a proxy for http/https access.

**Comment by gjsjohnmurray** [comment]
@betizad which driver are you using to connect to your database(s)?

**Comment by betizad** [comment]
@gjsjohnmurray  I'm using databricks sql driver.

**Comment by gjsjohnmurray** [comment]
> @gjsjohnmurray I'm using databricks sql driver.

Please raise your issue at https://github.com/databricks/sqltools-databricks-driver/issues
